src/devices/Nmea0183/SentenceCache.cs

```diff
@@ -293,6 +330,43 @@ public void Add(NmeaSentence sentence)
             OnNewSequence(null, sentence);
         }
 
+        /// <summary>
+        /// Adds the given sentence explicitly to the cache
+        /// </summary>
+        /// <param name="source">The source of the sentence</param>
+        /// <param name="sentence">The sentence</param>
+        public void Add(NmeaSinkAndSource source, NmeaSentence sentence)
+        {
+            OnNewSequence(source, sentence);
+        }
+
+        /// <summary>
+        /// Adds a sentence to the cache, but only to the list of sentences from this particular source,
+        /// so the message will not override the effect of a <see cref="GetLastSentence(Iot.Device.Nmea0183.SentenceId)"/> call.
+        /// </summary>
+        /// <param name="source">The source. Must be non-null</param>
+        /// <param name="sentence">The sentence. Must be non-null</param>
+        public void AddFromSource(NmeaSinkAndSource source, NmeaSentence sentence)
```

Comment:
Does user need to understand cache version of this method? Could this be maybe overload or different name or better description?

Reply:
I hope the description is clear enough.